main/src/ca/uwaterloo/flix/language/ast/Ast.scala

```diff
@@ -374,7 +374,7 @@ object Ast {
   /**
     * Represents a label in the labelled graph.
     */
-  case class Label(pred: Name.Pred, arity: Int)
+  case class Label(pred: Name.Pred, lattice: Boolean, terms: List[Type])
```

Comment:
- Use Ast.Denotation instead of the Boolean.
- I would add `arity: Int`. Yes its duplicate of terms, but I think it would make everything more explicit.